source/MRViewer/MRRenderMeshObject.h

```diff
@@ -21,16 +23,53 @@ class RenderMeshObject : public IRenderObject
     const ObjectMeshHolder* objMesh_;
 
     // need this to use per corner rendering (this is not simple copy of mesh vertices etc.)
-    mutable Buffer<Vector3f> vertPosBufferObj_;
-    mutable Buffer<Vector3f> vertNormalsBufferObj_;
-    mutable Buffer<Color> vertColorsBufferObj_;
-    mutable Buffer<UVCoord> vertUVBufferObj_;
-    mutable Buffer<Vector3i> facesIndicesBufferObj_;
-    mutable Buffer<Vector2i> edgesIndicesBufferObj_;
-    mutable Buffer<unsigned> faceSelectionTexture_;
-    mutable Buffer<Vector4f> faceNormalsTexture_;
-    mutable std::vector<Vector3f> borderHighlightPoints_;
-    mutable std::vector<Vector3f> selectedEdgesPoints_;
+    enum BufferType {
+        VERTEX_POSITIONS,
+        VERTEX_NORMALS,
+        FACE_NORMALS,
+        VERTEX_COLORMAPS,
+        FACES,
+        EDGES,
+        VERTEX_UVS,
+        FACE_SELECTION,
+        BORDER_LINES,
+        EDGE_SELECTION,
+
+        BUFFER_COUNT,
+    };
+    mutable Buffer<std::byte> bufferObj_;
+
+    template <BufferType>
+    struct ElementType;
+
+    mutable std::array<std::size_t, BUFFER_COUNT> elementCount_;
+    mutable std::bitset<BUFFER_COUNT> elementDirty_;
```

Comment:
try use uint64_t bindDirtyMask_

Reply:
I see that you use reference inside this bitset below. Let us add a name for the data type:
```
using DirtyFlags = std::bitset<BUFFER_COUNT>;

Reply:
Rewritten to reuse existing bit mask.